cloudprem/database.tf

```diff
@@ -298,20 +288,23 @@ resource "aws_dms_replication_task" "this" {
 }
 
 # AWS provider issue to replace this https://github.com/hashicorp/terraform-provider-aws/issues/2083
-# resource "null_resource" "start_replicating" {
-#   count = var.enable_bi ? 1 : 0
-
-#   triggers = {
-#     dms_task_arn = aws_dms_replication_task.this[0].replication_task_arn
-#   }
-
-#   provisioner "local-exec" {
-#     when    = create
-#     command = "aws dms start-replication-task --start-replication-task-type start-replication --replication-task-arn ${self.triggers["dms_task_arn"]}"
-#   }
-
-#   provisioner "local-exec" {
-#     when    = destroy
-#     command = "aws dms stop-replication-task --replication-task-arn ${self.triggers["dms_task_arn"]}"
-#   }
-# }
\ No newline at end of file
+ resource "null_resource" "replication_control" {
+   count = var.enable_bi ? 1 : 0
+
+   depends_on = [module.replicated]
```

Comment:
Interested. This seems like graph hacking.

Maybe we could leave a comment here explaining the connection? Some like:
```
   # Once the replicated module is running, we know that DMS is ready to poke
```

Thoughts?

Reply:
This _is_ kind of graph hacking but it's not as bad as you might think. I added this dependency because I don't want DMS to start until the application is deployed and there is some data to replicate. I saw some issues where DMS was unable to pickup new schemas after the replication task was started. It's a bug in DMS but this is an attempt to prevent it from happening. It's effectively baking a race condition in but without visibility into replicated's status wrt the application deployment there's not really any other way to achieve it. Open to suggestions tho :)

Reply:
Oh, i see. That makes more sense. But does the `replicated` module actually wait for the app to spin up?

Reply:
No it doesn't. Hence the baked in race condition. It's better than nothing but it is certainly not deterministic. The helm chart is only concerned with "is the replicated scheduler installed successfully". The app could explode and it would still succeed.

Reply:
Oh. I see. That's no good.

Here's an idea, and a filthy hack...

What do you think about using a `local-exec` or the likes to poll the DB for the db-initialize to at least start? We could spinlock looking for the `onprem_guide` schema to exist. That way we'll know DMS has _something_ to copy?

Reply:
The problem with that is terraform will not have access to the mysql database from a laptop and I really don't want to add any terraform code that locks us out of deploying these stacks on our workstations if possible. Same reason why we have to use that bastion user data hack to create the frontegg schemas. The "correct" answer is probably to create a lambda that will actually poll the app endpoint and hold the DMS start until it shows up but that's a lot more work for a relatively minor issue. After all if the BI database is not populated then the customer will know that and can just restart the replication task manually.

Reply:
Alright. So as long as this fails in an acceptable way under the worst case, then it's probably fine. This is probably "close enough" for most updates?